test/test_distributed_models.jl

```diff
@@ -0,0 +1,495 @@
+using MPI
+
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+using Oceananigans.Distributed: index2rank, east_halo, west_halo, north_halo, south_halo, top_halo, bottom_halo
+
+# Right now just testing with 4 ranks!
+comm = MPI.COMM_WORLD
+mpi_ranks = MPI.Comm_size(comm)
+@assert mpi_ranks == 4
+
+#####
+##### Multi architectures and rank connectivity
+#####
+
+function run_triply_periodic_rank_connectivity_tests_with_411_ranks()
+    topo = (Periodic, Periodic, Periodic)
+    full_grid = RegularRectilinearGrid(topology=topo, size=(8, 8, 8), extent=(1, 2, 3))
+    arch = MultiCPU(grid=full_grid, ranks=(4, 1, 1))
+
+    my_rank = MPI.Comm_rank(MPI.COMM_WORLD)
+    @test my_rank == index2rank(arch.my_index..., arch.ranks...)
+
+    connectivity = arch.connectivity
+
+    # No communication in y and z.
+    @test isnothing(connectivity.south)
+    @test isnothing(connectivity.north)
+    @test isnothing(connectivity.top)
+    @test isnothing(connectivity.bottom)
+
+    # +---+---+---+---+
+    # | 0 | 1 | 2 | 3 |
+    # +---+---+---+---+
+
+    if my_rank == 0
```

Comment:
Is this hard coding a split in 4?  For a test this is fine but does this work for other integers too?

Reply:
Yes. I've written the code to support any number of ranks but right now the tests actually only run using 4 ranks.

I asked around about how to best test multiple ranks, e.g. run tests with 2 ranks, then 4 ranks, then 8 ranks, etc. and it seemed like the recommended approach is to use separate Julia/MPI processes for this so we might end up having to hard-code some stuff =/

Reply:
Makes sense and glad you got it working